initramfs/init.sh

```diff
@@ -101,25 +103,44 @@ find_and_run_update()
 			continue
 		fi
 
-		if [ ! -x "${UPDATE_SCRIPT}" ]; then
+		if [ ! -x "${TOOLBOX_IMAGE}" ]; then
 			umount "${dev}"
-			echo "No executable update '${UPDATE_SCRIPT}' found on ${dev}, trying next."
+			echo "No update toolbox image '${TOOLBOX_IMAGE}' found on '${dev}', trying next."
 			continue
 		fi
 
-		echo "Found update on ${dev}, executing update ${UPDATE_SCRIPT}."
-		if ! "${UPDATE_SCRIPT}"; then
-			umount "${dev}"
-			echo "Update failed!"
+		echo "Found '${TOOLBOX_IMAGE}' on '${dev}', attempting to mount."
+		if ! mount "${TOOLBOX_IMAGE}" "${TOOLBOX_MOUNT}"; then
+			echo "Update failed: Unable to mount '${TOOLBOX_IMAGE}'."
 			critical_error
 			break;
 		fi
 
-		echo "Update finished, cleaning up."
-		if ! chmod -x "${UPDATE_SCRIPT}" || [ -x "${UPDATE_SCRIPT}" ]; then
-			umount "${dev}"
-			echo "Please remove update medium and power off."
-			shutdown
```

Comment:
note that while your future patch addresses this (ordering them in reverse) address this breaks bisectable behavior. just an FYI